e2e/test/helpers/TestDevice.cs

```diff
@@ -140,32 +81,70 @@ public string ConnectionString
 
+        public IotHubDeviceClient DeviceClient { get; private set; }
+
         public IAuthenticationMethod AuthenticationMethod { get; private set; }
 
-        public IotHubDeviceClient CreateDeviceClient(IotHubClientOptions options = default, ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device)
+        public async Task<IotHubDeviceClient> CreateDeviceClientAsync(IotHubClientOptions options = default, ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device, bool openClient = false)
         {
-            IotHubDeviceClient deviceClient = null;
-
             if (AuthenticationMethod == null)
             {
                 if (authScope == ConnectionStringAuthScope.Device)
                 {
-                    deviceClient = new IotHubDeviceClient(ConnectionString, options);
-                    VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClient)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from device connection string");
+                    DeviceClient = new IotHubDeviceClient(ConnectionString, options);
+                    VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClientAsync)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from device connection string");
                 }
                 else
                 {
-                    deviceClient = new IotHubDeviceClient($"{TestConfiguration.IotHub.ConnectionString};DeviceId={Device.Id}", options);
-                    VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClient)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from IoTHub connection string");
+                    DeviceClient = new IotHubDeviceClient($"{TestConfiguration.IotHub.ConnectionString};DeviceId={Device.Id}", options);
+                    VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClientAsync)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from IoTHub connection string");
                 }
             }
             else
             {
-                deviceClient = new IotHubDeviceClient(IotHubHostName, AuthenticationMethod, options);
-                VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClient)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from IAuthenticationMethod");
+                DeviceClient = new IotHubDeviceClient(IotHubHostName, AuthenticationMethod, options);
+                VerboseTestLogger.WriteLine($"{nameof(CreateDeviceClientAsync)}: Created {nameof(IotHubDeviceClient)} {Device.Id} from IAuthenticationMethod");
             }
 
-            return deviceClient;
+            if (openClient)
+            {
+                await OpenWithRetryAsync().ConfigureAwait(false);
+            }
+
+            return DeviceClient;
+        }
+
+        /// Sometimes the just newly created device can fail to connect if there is lag in the hub for the identity existing.
+        /// Retry a few times to prevent test failures.
+        /// </summary>
+        public async Task OpenWithRetryAsync()
```

Comment:
I moved this open with retry logic to TestDevice so it can be used by more tests.